### ISSUE TITLE
TFT's InterpretableMultiHeadAttention input

### ISSUE THREAD

**Issue by czarobxm** [jdb78/pytorch-forecasting#1444]
- PyTorch-Forecasting version: 1.0.0

In TFT's forward function an attention layer looks like this:
"""
# Attention
        attn_output, attn_output_weights = self.multihead_attn(
            q=attn_input[:, max_encoder_length:],  # query only for predictions
            k=attn_input,
            v=attn_input,
            mask=self.get_attention_mask(encoder_lengths=encoder_lengths, decoder_lengths=decoder_lengths),
        )
"""
Why the query is sliced to be only a decoder part of the input. Doesn't attention learn dependencies between all inputs? In Google's implementation (https://github.com/google-research/google-research/blob/master/tft/libs/tft_model.py#L994C11-L994C11) query is not sliced. Which version is right?
